must-gather/collection-scripts/gather

```diff
@@ -1,40 +1,4 @@
 #!/bin/bash
-BASE_COLLECTION_PATH="/must-gather"
-mkdir -p ${BASE_COLLECTION_PATH}
 
-echo "collecting operator pod logs" | tee -a  ${BASE_COLLECTION_PATH}/gather-debug.log
-operatorPodLogCollectionPath="${BASE_COLLECTION_PATH}/namespaces"
-operatorPodLogCollectionJSONPath="{range .items[*]}mkdir -p ${operatorPodLogCollectionPath}/{@.metadata.namespace}/logs/{@.metadata.name};oc logs {@.metadata.name}  --all-containers -n {@.metadata.namespace} &> ${operatorPodLogCollectionPath}/{@.metadata.namespace}/logs/{@.metadata.name}/{@.metadata.name}.log;{end}"
-operatorPodPreviousLogCollectionJSONPath="{range .items[*]}mkdir -p ${operatorPodLogCollectionPath}/{@.metadata.namespace}/logs/{@.metadata.name};oc logs {@.metadata.name} -p --all-containers -n {@.metadata.namespace} &> ${operatorPodLogCollectionPath}/{@.metadata.namespace}/logs/{@.metadata.name}/{@.metadata.name}-previous.log;{end}"
-{
-    timeout 120 oc get pods --all-namespaces -l 'name in (local-storage-operator,ocs-operator)' -o jsonpath="${operatorPodLogCollectionJSONPath}";
-    timeout 120 oc get pods --all-namespaces -l 'name in (local-storage-operator,ocs-operator)' -o jsonpath="${operatorPodPreviousLogCollectionJSONPath}";
-    timeout 120 oc get pods --all-namespaces -l 'app in (noobaa,rook-ceph-operator)' -o jsonpath="${operatorPodLogCollectionJSONPath}";
-    timeout 120 oc get pods --all-namespaces -l 'app in (noobaa,rook-ceph-operator)' -o jsonpath="${operatorPodPreviousLogCollectionJSONPath}";
-} >> collector.sh 2>>${BASE_COLLECTION_PATH}/gather-debug.log
-chmod +x collector.sh
-{ ./collector.sh; } >> ${BASE_COLLECTION_PATH}/gather-debug.log 2>&1
-# Resource List
-resources=()
-# collect storagecluster resources
-resources+=(storageclusters)
-
-# collect OB/OBC resoureces
-resources+=(objectbucketclaims)
-resources+=(objectbuckets)
-
-# Add general resources to list if necessary 
-
-# Run the Collection of Resources using must-gather
-for resource in "${resources[@]}"; do
-    echo "collecting dump of ${resource}" | tee -a  ${BASE_COLLECTION_PATH}/gather-debug.log
-    { timeout 120 oc adm --dest-dir=${BASE_COLLECTION_PATH} inspect "${resource}" --all-namespaces; } >> ${BASE_COLLECTION_PATH}/gather-debug.log 2>&1
-done
-
-# Call other gather scripts
-gather_noobaa_resources ${BASE_COLLECTION_PATH}
-gather_ceph_resources ${BASE_COLLECTION_PATH}
-
-echo "deleting empty files" >> ${BASE_COLLECTION_PATH}/gather-debug.log
-find "${BASE_COLLECTION_PATH}" -empty -delete >> ${BASE_COLLECTION_PATH}/gather-debug.log 2>&1
-exit 0
\ No newline at end of file
+# calling gather.sh from this file as must-gather basically looks for gather file in the path
```

Comment:
Is this change just so we have a file ending in `.sh` so the shellcheck test can find it? If so, please expand the comment to explain that as well.

Reply:
Yeah, that is a little weird.
If we need to have scripts not ending in ".sh", we should probably do something else than searching for `.sh` files, like grepping for `#!/bin/bash` or similar. E.g.:

```
find . \( -path ./vendor -o -path ./.git -o -path ./build \) -prune -o -name *~ -prune -o -type f -exec grep -l -e '^#!/bin/bash$' {} \;
```

That would avoid all the renaming...

Reply:
added the complete comment.